### ISSUE TITLE
Find better UI implementation for WP upload

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#8790]
Currently the upload to personal note for user defined waypoints is placed between the existing buttons for Edit and Upload. I do see some problems with that:
- As the buttons are already rather small the users might unintentionally click the wrong button
- Users might not understand the function of this button as the page, where it is shown is unrelated to the data source used for it

On the other hand users will currently easily find the new function, but still I would like to suggest to find an alternative place for that function in the UI.

How about a third button below "Add current location" on top of the waypoint tab of a cache?
This would also have the advantage that longer strings fit in, such as "Copy user modified waypoints to personal note", which makes the function more self explanatory.

@eddiemuc

**Comment by eddiemuc** [comment]
Hi @Lineflyer , I placed the button near the PN (not the WPs) for multiple reasons:
* I wanted the User to see directly the implication of using this function (e.g. how the PN is modified)
* I think of this function primarily as a backup function, so naturally a User would choose "Upload" right after storing WP information - and "Upload" is placed near the PN
* In the future I would like to extend this function to also store other Cache-Information (Not only WPs) in the PN for Backup and restoring. For example I would Like to Store the list names n which the Cache is stored (maybe using # as a Marker). (No Feature request issued yet)

In summary, I think this function should not be WP-centric but PN-centric in the UI.

This is of course debatable. What is your View in this?

**Comment by MagpieFourtyTwo** [comment]
I see your point, nevertheless I go with @Lineflyer regarding the point

> As the buttons are already rather small the users might unintentionally click the wrong button

as this happened annoyingly frequently this weekend (while we did a bunch of complex Multis and Mysteries).

Perhaps you may take into consideration a coming up feature request of mine: Although the new sizing of the PN (popup grows on adding lines) catapults working with PNs to a completely new level, the currently used popup behaves kind of awkward now, and I guess, changing it from popup to a dedicated (replacing) window may solve all of them in one go. So I will suggest to change this popup into a real window, like the log window is. 

Maybe then there is even more space at the bottom, to put buttons for this into there ...?

**Comment by moving-bits** [comment]
> Currently the upload to personal note for user defined waypoints is placed between the existing buttons for Edit and Upload. I do see some problems with that:
> As the buttons are already rather small the users might unintentionally click the wrong button

I agree with that so far, but understand @eddiemuc's arguments as well:
> In summary, I think this function should not be WP-centric but PN-centric in the UI.

Therefore I propose a different solution for positioning the third button:
Why not put the button below the personal note, but left aligned, in the same row as the "edit" button?

So you would have something like this:

|Personal|note|text|
|----|-----|-----|
| Store WP | | Edit |
| | | Upload|

(table to be replaced by image when I'm back at my PC)

**Comment by eddiemuc** [comment]
I Support @moving-bits s idea

**Comment by MagpieFourtyTwo** [comment]
Me2. Was my first idea, thought I already wrote it, but probably forgot to submit the message. ;)

**Comment by eddiemuc** [comment]
I can implement it next week hopefully

**Comment by MagpieFourtyTwo** [comment]
> So you would have something like this:
> Personal note text
> Store WP _________ Edit
> ___________________ Upload

Correct me if I'm wrong, but to me it seems as if on second press of `` Store waypoints`` the data will be written back from PN to the WP list - correct?  That was at least my impression while testing: After first press of ``Store waypoints`` I changed the type of the "Final" WP to "Reference Point", but this change got immediately overwritten and changed back to "Final" after I pressed ``Store Waypoints`` a second time.

*If* I'm right and *if* this is intended, I would highly recommend a second button ``Restore WPs`` below the ``Store WPs``, to unmistakably make clear, which action will happen,

**Comment by moving-bits** [comment]
Which would then give us a UI like this:

![image](https://user-images.githubusercontent.com/3754370/90066660-53887380-dcee-11ea-868c-6821cc0fb3f8.png)

(This is from my veeery-long-personal-note test case)

**Comment by MagpieFourtyTwo** [comment]
YEAH! ;)

**Comment by eddiemuc** [comment]
> Correct me if I'm wrong, but to me it seems as if on second press of `` Store waypoints`` the data will be written back from PN to the WP list - correct? 

No, this action should only ever rewrite the PN from the WP list. But immediately after, the normal parsing of PNs for waypoints is triggered. The behavior you see might be a bug in the second step. If behavior persists maybe it would be good to Open a separate issue

**Comment by MagpieFourtyTwo** [comment]
Ok, will keep this in mind.

**Comment by Lineflyer** [comment]
I also like @moving-bits idea.
But we do not need the restore waypoints button as this is still served by the existing coordinate parser for PN, meaning they will automatically be restored by just the presence of the backup in the PN, which is a really nice approach as I can e.g. just open the same cache on another device with c:geo running on my account and have the same WP status.
The user can however optionally use the existing "stop extraction" switch to prevent it.

**Comment by eddiemuc** [comment]
#8817 implements the GUI as requested. There are now two buttons:
* Add/update waypoints: functionality of current "store waypoints" button
* Remove waypoints: removes previously created "waypoint block" from notes, if there is any
